### ISSUE TITLE
WAR deploy fails with error

### ISSUE THREAD

**Issue by DrWala** [microsoft/azure-pipelines-tasks#11776]
## Note  
Issues in this repo are for tracking bugs, feature requests and questions for the tasks in this repo 

For a list:  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

If you have an issue or request for the Azure Pipelines service, use developer community instead:  

https://developercommunity.visualstudio.com/spaces/21/index.html )

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: Deploy Azure App Service (for what its worth it doesn't work with AzureRmWebAppDeploymentV4 either)

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines or TFS on-premises?
    
    - If using TFS on-premises, provide the version: 
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: nyx-intelligence, cosmomo, 

- Agent - Hosted or Private: 
    
    - If using Hosted agent, provide agent queue name: vs2017-win2016 (is this right?)

    - If using private agent, provide the OS of the machine running the agent and the agent version: 

## Issue Description

[Include task name(s), screenshots and any other relevant details]
War deploy to app service on linux consistently results in the following error:

`Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory.`

### Task logs

[Enable debug logging and please provide the zip file containing all the logs for a speedy resolution]
[logs_125.zip](https://github.com/microsoft/azure-pipelines-tasks/files/3849899/logs_125.zip)

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

Build a maven project and then deploy that to an azure web app on linux

All tasks: 
![image](https://user-images.githubusercontent.com/6803997/68918667-25e7bb80-07aa-11ea-8755-6c79d758629a.png)

### Error logs

```
##[debug]setting affinity cookie ["ARRAffinity=542cd8f86e22b05d0dda6b1e769f7515c08dfb72220a509d16cc7ae3286450a8;Path=/;HttpOnly;Domain=cosmomo-backend-dev.scm.azurewebsites.net"]
##[debug][GET]https://cosmomo-backend-dev.scm.azurewebsites.net/api/deployments/temp-9ce760bd/log/cdbff7df-ecd4-4dee-8b32-7115b557d6f7
##[debug]getDeploymentLogs. Data: {"statusCode":200,"statusMessage":"OK","headers":{"transfer-encoding":"chunked","content-type":"application/json; charset=utf-8","server":"Kestrel","date":"Fri, 15 Nov 2019 05:11:59 GMT","connection":"close"},"body":[{"log_time":"2019-11-15T05:11:48.1581561Z","id":"","message":"Cleaning up temp folders from previous zip deployments and extracting pushed zip file /tmp/zipdeploy/7d69c57b-bdfb-44e9-bd11-e87869b367aa.zip (71.07 MB) to /tmp/zipdeploy/extracted","type":0,"details_url":null},{"log_time":"2019-11-15T05:11:48.190732Z","id":"","message":"Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory.","type":2,"details_url":null}]}
Cleaning up temp folders from previous zip deployments and extracting pushed zip file /tmp/zipdeploy/7d69c57b-bdfb-44e9-bd11-e87869b367aa.zip (71.07 MB) to /tmp/zipdeploy/extracted
Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory.
##[error]Failed to deploy web package to App Service.
##[debug]Processed: ##vso[task.issue type=error;]Failed to deploy web package to App Service.
##[debug]Deployment Failed with Error: Error: Package deployment using ZIP Deploy failed. Refer logs for more details.
##[debug]task result: Failed
##[error]Error: Package deployment using ZIP Deploy failed. Refer logs for more details.
```

**Comment by 20shivangi** [comment]
@DrWala Can you please attach the complete debug logs by setting system.debug variable to true in your pipeline?

**Comment by DrWala** [comment]
I think this issue can be closed, it started working a few days after I posted this but forgot to comment. I was playing with azure devops for the first time essentially, and I got the war package deployed after making a few changes in the pom.xml—namely changing the packaging type to war, and testing on a local tomcat server after building the package manually using maven package. Once I got that working the deployment through the pipelines worked as well. So I suppose this was user error more than anything else, although the lack of documentation on this is definitely frustrating.
